### PR TITLE
vsphere: Use a fork of vmware/vic

### PR DESCRIPTION
**Pull request by notnoop**
Use a fork of github.com/vmware/vic that uses
github.com/sirupsen/logrus:
https://github.com/hashicorp/vic/commits/b15179225d21a06e2a9e7bfcfc6d2e9225c978fb
.

Incidentally, update the logrus dependency in vendor to v1.0.6 to match
the one used by go mod

Fixes: https://github.com/hashicorp/go-discover/issues/94 .

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -70,3 +69,5 @@ require (
 	k8s.io/apimachinery v0.0.0-20180821005732-488889b0007f
 	k8s.io/client-go v8.0.0+incompatible
 )
+
+replace github.com/vmware/vic => github.com/hashicorp/vic v1.5.1-0.20190215174116-b15179225d21
```
Comment (alvin-huang) [inline go.mod]: Will this work? I don't see your previous changes in `hashicorp/vic`

Comment (rboyer) [reply go.mod]: I thought that `replace` only works for the "main module" aka the thing being built (like `consul` or `nomad`). From https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive:
> As described in the 'go.mod' concepts section above, replace directives provide additional control in the top-level go.mod for what is actually used to satisfy a dependency found in the Go source or go.mod files, **while replace directives in modules other than the main module are ignored when building the main module.**

That bolded part above implies that you would need to copy this replace snippet into the `go.mod` files of everything that directly or indirectly uses `go-discover`

Comment (notnoop) [reply go.mod]: good point @rboyer  - I confirmed the behavior you referenced too.  I guess we must use reference the fork.

Comment (notnoop) [reply go.mod]: Updated to "hard link" vmware/vic instead.  Thanks for catching this.

**Comment by banks** [comment]
I've updated this PR to use the master branch of our fork at a specific commit rather than a fake version tag that might collide with an upstream one eventually.

The embedded vendor dirs are really confusing...

I tested that the vsphere provider (only part touched here) still builds and runs tests both with go-mod on and off:

```bash
$ env GO111MODULE=on go build ./...
# no output which is good
$ cd provider/vsphere/
$ env GO111MODULE=off go test
PASS
ok  	github.com/hashicorp/go-discover/provider/vsphere	0.274s
$ env GO111MODULE=on go test
PASS
ok  	github.com/hashicorp/go-discover/provider/vsphere	0.226s
$ cd ../..
# Sanity check no upper case Sirupsen any more
$ rg  'sirupsen' | wc -l
      28
$ rg  'Sirupsen' | wc -l
       0
$ go mod graph | rg 'irupsen/logrus'
github.com/hashicorp/go-discover github.com/sirupsen/logrus@v1.0.6
```

And also that the whole project builds with go mod on:

It doesn't build with go mod off currently because the k8s provider has missing vendor deps. I'm not going to try and fix that here.

**Comment by banks** [comment]
@alvin-huang the CI failures seem to be 403 is this a known issue - they are unrelated to this PR anyway.

**Comment by banks** [comment]
Going to merge this as the CI failures are not related to changes made here.

**Pull request merged**